forest/cli/src/cli/send_cmd.rs

```diff
@@ -7,9 +7,75 @@ use forest_json::message::json::MessageJson;
 use forest_rpc_client::{mpool_push_message, wallet_default_address};
 use fvm_shared::{address::Address, econ::TokenAmount, message::Message, METHOD_SEND};
 use num::BigInt;
+use rust_decimal::prelude::*;
+use rust_decimal_macros::dec;
 
 use super::{handle_rpc_err, Config};
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct FILAmount {
+    pub value: TokenAmount,
+}
+
+impl FromStr for FILAmount {
+    type Err = anyhow::Error;
+
+    /// `FILAmount::from_str` parses a string slice and converts the amount into
+    /// an `attoFIL` `TokenAmount`, returning an error if the input: contains no
+    /// digits (e.g., `fil`), is not a valid number (e.g., `0.0.0fil`), is too
+    /// long (greater than `50` characters), is negative, or has an unrecognized
+    /// suffix. Fractional input values are valid unless the input units are
+    /// attoFIL (e.g., `1.234 attoFIL` is invalid while `1234 attoFIL` is valid
+    /// and parses to a `TokenAmount` corresponding to `1234` `attoFIL`). To
+    /// match the current behavior in Lotus, the default units are `FIL`
+    /// (e.g., `1` parses as `FIL` with a resulting `TokenAmount` corresponding
+    /// to `1e18` `attoFIL`). The function also accepts all units currently
+    /// supported by forest wallet in the `format_balance_string` function
+    /// (`attoFIL`, `femtoFIL`, `picoFIL`, `nanoFIL`, `microFIL`, `milliFIL`,
+    /// and `FIL`), and the output of `FILAmount::from_str` can be used as input
+    /// to the `format_balance_string` function.
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
```

Comment:
- This documentation is attached to `FILAmount::from_str` so you don't need to refer back to the function.
- I don't know what this means: "an attoFIL TokenAmount".
- It's best if you cover the positive definition first and then briefly cover the negative definition as a separate paragraph.
- This is not correct: "Fractional input values are valid unless the input units are attoFIL". For example, this input is invalid "0.0001 femto fil". That is because "0.0001 femto fil" is the same as "0.1 atto fil".
- The note about Lotus isn't wrong but it's not the only behaviour that matches Lotus. It's shorter and more precise: We accept all input strings accepted by Lotus.
- You got the relationship between `FILAmount::from_str` and `format_balance_string` the wrong way around. The type-system statically guarantees that the output of `FILAmount::from_str` can be used by `format_balance_string`. The interesting relationship is whether the output of `format_balane_string` can be parsed by `FILAmount::from_str`.
- In general, text documentation should be short and to-the-point.

Try something like this:

Parse a string like `10 attoFIL` or `0.5 milliFIL` into a `TokenAmount`. Known units are `aFIL`, `attoFIL`, `femtoFIL`, `picoFIL`, `nanoFIL`, `microFIL`, and `milliFIL`. These are the same units used by `format_balance_string`.

The `FIL` suffix may be omitted. The parsed `TokenAmount` may not be negative, contain a fractional number of attoFIL, or be longer than 50 digits. Any input parseable by Lotus is considered to be valid.

Examples:
```rust
// Without a unit, numbers are interpreted as FIL
assert_eq!(
  FILAmount::from_str("1.5").unwrap(),
  FILAmount::from_str("1.5 FIL").unwrap(),
);
```

```rust
// aFIL is a synonym of attoFIL
assert_eq!(
  FILAmount::from_str("10 attoFIL").unwrap(),
  FILAmount::from_str("10 aFIL").unwrap(),
);
```

```rust
// Suffixes are case insensitive and may omit `FIL`
assert_eq!(
  FILAmount::from_str("0.5 milli").unwrap(),
  FILAmount::from_str("0.5 MiLlI FIL").unwrap(),
);
```
```rust
// Negative values are invalid
assert!(FILAmount::from_str("-0.5 FIL").is_err());
```

```rust
// Fractional attoFIL amounts are invalid
assert!(FILAmount::from_str("0.0001 picoFIL").is_err());
```